lib/rom/adapter/lint/test.rb

```diff
@@ -0,0 +1,83 @@
+module ROM
+  class Adapter
+    module Lint
+      # This is a simple lint-test for an adapter class to ensure the basic
+      # interfaces are in place
+      #
+      # @example
+      #
+      #   class MyAdapterTest < MiniTest::Test
+      #     include ROM::Adapter::Lint::TestAdapter
+      #
+      #     def setup
+      #       @adapter = MyAdapter
+      #       @uri = "super_db://something"
+      #     end
+      #   end
+      #
+      # @public
+      module TestAdapter
+        attr_reader :adapter, :uri
+
+        def test_schemes
+          assert_respond_to adapter, :schemes,
+            "#{adapter}.schemes must be implemented"
+
+          assert_instance_of Array, adapter.schemes,
+            "#{adapter}.schemes must return an array with supported URI schemes"
+
+          assert adapter.schemes.size > 0,
```

Comment:
@solnic 

With `minitest` we can use `assert_operator adapter.schemes.size, :>, 0`.
It gives a nice error message IMHO.

Alternatively, we could also use `assert adapter.schemes.any?` to match the assert message better?